charts/nginx-gateway-fabric/values.yaml

```diff
@@ -493,6 +493,18 @@ nginx:
     # - port: 30025
     #   listenerPort: 80
 
+  ## Defines the settings for the data plane readiness probe. This probe returns Ready when the NGINX data plane is ready to serve traffic.
+  readinessProbe: {}
+    # @schema
+    # type: integer
+    # minimum: 1
+    # maximum: 65535
+    # @schema
+    # -- Port in which the readiness endpoint is exposed.
+    # port: 8081
+
+    # The Kubernetes probe can also be configured by modifying the NginxProxy resource.
+
```

Comment:
Let's just put this under the container spec as well, it doesn't need to be at top level field. Then in the yaml, you shouldn't need to add it since it'll be added automatically in the container spec.